client/ayon_core/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,722 @@
+                        # popping last thumbnail from list since it is only one
+                        # and we do not need to iterate again over it
+                        if not relative_thumbnail_path:
+                            relative_thumbnail_path = thumbnails.pop()
+                        thumb_dir, thumb_file = \
+                            self._get_refactor_thumbnail_path(
+                                staging_dir, relative_thumbnail_path)
+                        _, ext = os.path.splitext(thumb_file)
+                        thumbnail_repr_data = deepcopy(
+                            base_thumbnail_repre_data)
+                        thumbnail_repr_data.update({
+                            "ext": ext[1:],
+                            "files": thumb_file,
+                            "stagingDir": thumb_dir
+                        })
+                        new_instance["prepared_data_for_repres"].append({
+                            "type": "thumbnail",
+                            "colorspace": None,
+                            "representation": thumbnail_repr_data,
+                        })
+                        # also add thumbnailPath for ayon to integrate
+                        if not new_instance.get("thumbnailPath"):
+                            new_instance["thumbnailPath"] = (
+                                os.path.join(thumb_dir, thumb_file)
+                            )
+
+                        thumbnails_processed = True
+
+                    # get representation data
+                    representation_data = self._get_representation_data(
+                        filepath, repre_data, staging_dir,
+                        explicit_output_name
+                    )
+
+                    new_instance["prepared_data_for_repres"].append({
+                        "type": "media",
+                        "colorspace": repre_data["colorspace"],
+                        "representation": representation_data,
+                    })
+
+    def _get_refactor_thumbnail_path(
+            self, staging_dir, relative_thumbnail_path):
+        thumbnail_abs_path = os.path.join(
+            staging_dir, relative_thumbnail_path)
+        return os.path.split(
+            thumbnail_abs_path)
+
+    def _get_representation_data(
+        self, filepath, repre_data, staging_dir, explicit_output_name=None
+    ):
+        """Get representation data
+
+        Args:
+            filepath (str): Filepath to representation file.
+            repre_data (dict): Representation data from CSV file.
+            staging_dir (str): Staging directory.
+            explicit_output_name (Optional[str]): Explicit output name.
+                For grouping purposes with reviewable components.
+                Defaults to None.
+        """
+
+        # get extension of file
+        basename = os.path.basename(filepath)
+        _, extension = os.path.splitext(filepath)
```

Comment:
To be sure both `.PNG` and `.png` are valid.
```suggestion
        extension = os.path.splitext(filepath)[-1].lower()
```

Reply:
added